### ISSUE TITLE
Crash mid-sync, repeatable

### ISSUE THREAD

**Issue by h110hawk** [vsespb/mt-aws-glacier#21]
I got this last night after having to control-c out of the upload. I rebooted "just in case" and there are no other updates or changes. I am on an unmodified pull of the release branch. What information do you need from me?

perl -v:
This is perl 5, version 12, subversion 4 (v5.12.4) built for darwin-thread-multi-2level

macintosh-2:mt-aws-glacier kelly$ port list installed | grep p5.12
p5.12-encode-locale            @1.20.0         perl/p5-encode-locale
p5.12-file-listing             @6.20.0         perl/p5-file-listing
p5.12-html-form                @6.0.0          perl/p5-html-form
p5.12-html-parser              @3.690.0        perl/p5-html-parser
p5.12-html-tagset              @3.200.0        perl/p5-html-tagset
p5.12-http-cookies             @6.10.0         perl/p5-http-cookies
p5.12-http-daemon              @6.10.0         perl/p5-http-daemon
p5.12-http-date                @6.20.0         perl/p5-http-date
p5.12-http-message             @6.60.0         perl/p5-http-message
p5.12-http-negotiate           @6.10.0         perl/p5-http-negotiate
p5.12-io-socket-ssl            @1.440.0        perl/p5-io-socket-ssl
p5.12-json-xs                  @2.230.0        perl/p5-json-xs
p5.12-libwww-perl              @6.40.0         perl/p5-libwww-perl
p5.12-lwp-mediatypes           @6.10.0         perl/p5-lwp-mediatypes
p5.12-lwp-protocol-https       @6.20.0         perl/p5-lwp-protocol-https
p5.12-mime-base64              @3.130.0        perl/p5-mime-base64
p5.12-mozilla-ca               @20110904       perl/p5-mozilla-ca
p5.12-net-http                 @6.10.0         perl/p5-net-http
p5.12-net-libidn               @0.120.0        perl/p5-net-libidn
p5.12-net-ssleay               @1.480.0        perl/p5-net-ssleay
p5.12-uri                      @1.590.0        perl/p5-uri
p5.12-www-robotrules           @6.10.0         perl/p5-www-robotrules

macintosh-2:mt-aws-glacier me$ git status
On branch master
 Untracked files:
(use "git add <file>..." to include in what will be committed)

config.cfg
nothing added to commit but untracked files present (use "git add" to track)

macintosh-2:mt-aws-glacier me$ ./mtglacier.pl sync --config config.cfg --journal ../Documents/glacier-logs/Joe-Laptop-Backup.log --from-dir ~/Desktop/jd/ --to-vault Joe-Laptop-Backup --concurrency=2
MT-AWS-Glacier, part of MT-AWS suite, Copyright (c) 2012  Victor Efimov http://mt-aws.com/ Version 0.82beta

PID 292 Started worker
PID 293 Started worker
Skip .DS_Store
...snip...
Found 24000 local files
PID 293 Created an upload_id 1BQOjEjLyqnQHdL0ePjyZiSWz5-ZPzgziccrkH1_jbjDXl71-8ksCY_Nd0fvGmfIFeFXCY7_HJIcEX7NnNafZ9sdUQZu
DIE outside EVAL block [0]
Call stack: 
ParentWorker::process_task(./mtglacier.pl:131)
ParentWorker::wait_worker(ParentWorker.pm:63)
JobListProxy::finish_task(ParentWorker.pm:84)
JobProxy::finish_task(JobListProxy.pm:80)
FileCreateJob::finish_task(JobProxy.pm:53)
FileUploadJob::new(FileCreateJob.pm:66)
main::**ANON**(FileUploadJob.pm:38)
Fatal Error: 0 Died at FileUploadJob.pm line 38, <GEN6> line 1.

**Comment by vsespb** [comment]
take a look at fileuploadjob line 38. this looks like a bug. file
modification time can be zero (y1970, epoch time) please modifty that file
so it should print the filename when die. then reproduce the bug and check
file modif. time manually. thanks.
 21.01.2013 0:55 ÐÏÌØÚÏ×ÁÔÅÌØ "Kelly Kane" notifications@github.com
ÎÁÐÉÓÁÌ:

**Comment by h110hawk** [comment]
Sure enough:

$VAR1 = bless( {
                 'partsize' => 16777216,
                 'filename' => '/Users/kelly/Desktop/jd/Downloads/attachments_2009_10_14/Arbitrary Influence.jpg',
                 'mtime' => 0,
                 'fh' => *{'FileCreateJob::$fh'},
                 'upload_id' => 'nrk1vO9VwY3k8A5sIyJa0UbzGuN0kq97jkfadffCCl1Stuq6739rpj06TV4mtyPlRGeqYlEUaFHVS-2E3n18FNPDb15G',
                 'relfilename' => 'Downloads/attachments_2009_10_14/Arbitrary Influence.jpg'
               }, 'FileUploadJob' );

**Comment by vsespb** [comment]
and is this file modification time indeed in y1970 ?
21.01.2013 5:45 ÐÏÌØÚÏ×ÁÔÅÌØ "Kelly Kane" notifications@github.com
ÎÁÐÉÓÁÌ:

**Comment by h110hawk** [comment]
Sorry, I was rushing to dinner. 

Yes, the mtime is indeed 0 (epoch). I `touch`'d the file to give it a non-zero mtime and the program worked on it and failed on the next mtime=0 one. Is the mtime=0 death required? Are you trying to detect when things change? If so mtime > 0 will still move time forward.

**Comment by vsespb** [comment]
it's a bug. ofcourse mtime 0 is valid value. i will fix in the evening.
21.01.2013 8:17 ÐÏÌØÚÏ×ÁÔÅÌØ "Kelly Kane" notifications@github.com
ÎÁÐÉÓÁÌ:

**Comment by vsespb** [comment]
So, this appear to be fixed. Thanks for reporting.

I store modification time among filename in Amazon Glacier description. So when files downloaded, modification time is restored.
When I first coded this I had support for mtime > 0 only, but very soon I realized that it can be also negative. Unfortunately I forgot to test case when it's = 0.

Now it's fixed (in v0.83beta, master branch) and I added several tests to avoid it in the future.
